Helper/Cart.php

```diff
@@ -1284,22 +1286,115 @@ public function getCartData($paymentOnly, $placeOrderPayload, $immutableQuote =
             $taxAmount = 0;
         }
 
+        $this->setLastImmutableQuote($immutableQuote);
+
         // include potential rounding difference and reset $diff accumulator
         $cart['items'][0]['total_amount'] += round($diff);
         $totalAmount += round($diff);
         $diff = 0;
 
         // add discount data
-        $cart['discounts'] = [];
+        list ($cart, $totalAmount, $diff) = $this->collectDiscounts (
+            $cart, $totalAmount, $diff, $immutableQuote, $address, $totals, $paymentOnly
+        );
 
+        /////////////////////////////////////////////////////////////////////////////////
+        // Add fixed amount type to all discounts if total amount is negative
+        // and set total to 0. Otherwise add calculated diff to cart total.
+        /////////////////////////////////////////////////////////////////////////////////
+        if ($totalAmount < 0) {
+            $totalAmount = 0;
+            foreach ($cart['discounts'] as &$discount) {
+                $discount['type'] = 'fixed_amount';
+            }
+        } else {
+            // add the diff to first item total to pass bolt order create check
+            $cart['items'][0]['total_amount'] += round($diff);
+            $totalAmount += round($diff);
+        }
+        /////////////////////////////////////////////////////////////////////////////////
+
+        $cart['total_amount'] = $totalAmount;
+        $cart['tax_amount']   = $taxAmount;
+
+        if (abs($diff) >= $this->treshold) {
```

Comment:
I know this is old code but I just noticed the typo. Can you fix this to `$this->threshold`?

Reply:
Done.